wallet/src/ui/app/components/network-switch/NetworkSelector.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useMemo, useCallback } from 'react';
+
+import { API_ENV_TO_INFO, API_ENV } from '_app/ApiProvider';
+import BsIcon from '_components/bs-icon';
+import { useAppSelector, useAppDispatch } from '_hooks';
+import {
+    setApiEnv,
+    changeRPCNetwork,
+    setNetworkSelector,
+} from '_redux/slices/app';
+import { getTransactionsByAddress } from '_redux/slices/txresults';
+import store from '_store';
+
+import st from './Network.module.scss';
+
+const NetworkSelector = () => {
+    const selectedApiEnv = useAppSelector(({ app }) => app.apiEnv);
+    const dispatch = useAppDispatch();
+    const networkList: any = API_ENV_TO_INFO;
+    const netWorks = useMemo(
+        () =>
+            Object.keys(networkList).map((itm: string) => ({
+                style: { color: networkList[itm].color },
+                ...networkList[itm],
+                networkName: itm,
+            })),
+        [networkList]
+    );
+
+    const changeNetwork = useCallback(
```

Comment:
I think we could use only `setApiEnv` and have a middleware to change the provider but also clear the state that is relative with the connected network. Currently there are cases we show coins or nfts that are from another network

Reply:
Not sure why we want to show assets from a different network at the same time as the current networks.

Reply:
I meant that with the current implementation we see objects that are in another network and we shouldn't show them. (eg when we change the network we see coins of the other network until we poll again and hope it is successful!)